python/pylibcugraph/pylibcugraph/k_core.pyx

```diff
@@ -0,0 +1,176 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from libc.stdint cimport uintptr_t
+import warnings
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_view_free,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.core_algorithms cimport (   
+    cugraph_core_result_t,
+    cugraph_k_core_result_t,
+    cugraph_core_result_create,
+    cugraph_k_core,
+    cugraph_k_core_degree_type_t,
+    cugraph_k_core_result_get_src_vertices,
+    cugraph_k_core_result_get_dst_vertices,
+    cugraph_k_core_result_get_weights,
+    cugraph_k_core_result_free,
+    cugraph_core_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+    create_cugraph_type_erased_device_array_view_from_py_obj,
+)
+
+def k_core(ResourceHandle resource_handle,
+           _GPUGraph graph,
+           size_t k,
+           degree_type,
+           core_result,
+           bool_t do_expensive_check):
+    """
+    Compute the k-core of the graph G
+    A k-core of a graph is a maximal subgraph that
+    contains nodes of degree k or more. This call does not support a graph
+    with self-loops and parallel edges.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resource needed for
+        referencing data and running algorithms.
+    
+    graph : SGGraph or MGGraph
+        The input graph, for either Single or Multi-GPU operations.
+    
+    degree_type: str
+        This option determines if the core number computation should be based
+        on input, output, or both directed edges, with valid values being
+        "incoming", "outgoing", and "bidirectional" respectively.
+        This option is currently ignored in this release, and setting it will
+        result in a warning.
+
+        This implementation only supports bidirectional edges.
+    
+    k : size_t (default=None)
```

Comment:
can you move this so it's ordered to match the order of the args in the signature?